src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -208,5 +212,343 @@ public void constantMacroWithUserSymbol() throws Exception {
         }
     }
 
-    // TODO additional tests
+    @Test
+    public void structMacroWithOneOptionalInvoked() throws Exception {
```

Comment:
We can make these tests less verbose over time, but I'd rather err on the side of verbose and debuggable at this point. As previously mentioned, the goal is not to write significantly more of these by hand, but rather to rely on the conformance test vectors long-term.